api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/generic/StartedArtMinusARTCareEnrollmentDateCalculationIMER1B.java

```diff
@@ -0,0 +1,155 @@
+import org.joda.time.Interval;
+import org.openmrs.Location;
+import org.openmrs.api.context.Context;
+import org.openmrs.calculation.patient.PatientCalculationContext;
+import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.eptsreports.metadata.HivMetadata;
+import org.openmrs.module.eptsreports.reporting.calculation.AbstractPatientCalculation;
+import org.openmrs.module.eptsreports.reporting.calculation.BooleanResult;
+import org.openmrs.module.eptsreports.reporting.utils.EptsCalculationUtils;
+import org.openmrs.module.reporting.data.patient.definition.SqlPatientDataDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.springframework.stereotype.Component;
+
+/** Returns patients who initiated ART within 15 days from the ART Care Enrollment */
+@Component
+public class StartedArtMinusARTCareEnrollmentDateCalculationIMER1B
+    extends AbstractPatientCalculation {
+
+  private static final String ON_OR_BEFORE = "onOrBefore";
+  private static final int INTERVAL_BETWEEN_ART_START_DATE_MINUS_PATIENT_ART_ENROLLMENT_DATE = 15;
+
+  @Override
+  public CalculationResultMap evaluate(
+      Collection<Integer> cohort,
+      Map<String, Object> parameterValues,
+      PatientCalculationContext context) {
+    CalculationResultMap map = new CalculationResultMap();
+    CalculationResultMap artStartDates =
+        calculate(
+            Context.getRegisteredComponents(InitialArtStartDateCalculation.class).get(0),
+            cohort,
+            parameterValues,
+            context);
+    CalculationResultMap preARTCareEnrollmentMap = getARTCareEnrollment(cohort, context);
+
+    Date endDate = (Date) parameterValues.get(ON_OR_BEFORE);
+
+    if (endDate == null) {
+      endDate = (Date) context.getFromCache(ON_OR_BEFORE);
+    }
+
+    if (endDate != null) {
+      for (Integer patientId : cohort) {
+        boolean lessOrEqualTo15Days = false;
+        Date artStartDate =
+            InitialArtStartDateCalculation.getArtStartDate(patientId, artStartDates);
+
+        SimpleResult preARTCareEnrollmentResult =
+            (SimpleResult) preARTCareEnrollmentMap.get(patientId);
+
+        if (preARTCareEnrollmentResult != null) {
+          Date preARTCareEnrollmentDate = (Date) preARTCareEnrollmentResult.getValue();
+
+          if (artStartDate != null
+              && preARTCareEnrollmentDate != null
+              && preARTCareEnrollmentDate.getTime() < artStartDate.getTime()) {
+            int days =
+                Days.daysIn(
+                        new Interval(preARTCareEnrollmentDate.getTime(), artStartDate.getTime()))
+                    .getDays();
+
+            if (days <= INTERVAL_BETWEEN_ART_START_DATE_MINUS_PATIENT_ART_ENROLLMENT_DATE) {
+              lessOrEqualTo15Days = true;
+            }
+          }
+        }
+
+        map.put(patientId, new BooleanResult(lessOrEqualTo15Days, this));
```

Comment:
please change BooleanResult to SimpleResult  to allow us to pass the interval between care start date and art start date

Reply:
Done